internal/plugin/connectors/tcp/mssql/connector.go

```diff
@@ -94,15 +96,108 @@ Overview of the connection process
 
 // SingleUseConnector is used to create an authenticated connection to an MSSQL target
 type SingleUseConnector struct {
-	backendConn net.Conn
-	clientConn  net.Conn
-	logger      log.Logger
+	backendConn       net.Conn
+	clientConn        net.Conn
+	logger            log.Logger
+	// Note: We're following standard ctor naming practices with this field.
+	newMSSQLConnector types.MSSQLConnectorCtor
+	readPrelogin      types.ReadPreloginFunc
```

Comment:
I should have thought of this before... for the `readPrelogin`, `writePrelogin`, and `readLogin` fields of the SingleUseConnector, would it be clearer to include whether the messages are request vs. acks, e.g. `readPreloginReq`, `writePreloginAck`, `readLoginReq`, or is it already clear that these apply on the client side, so request vs. ack is obvious?
When we merge this with Kumbi's message pass-through, I suspect we'll need to add a `writeLoginAck` (Or `writeLogin`).

Reply:
Yes, it would be clearer.  Let's wait till we merge with Kumbi though since we'll be making many other changes as well.